test/AutoRest.Shared.Tests/RequestContentHelperTests.cs

```diff
@@ -9,6 +9,62 @@ namespace Azure.Core.Tests
 {
     public class RequestContentHelperTests
     {
+        [TestCase(1, 2)]
+        [TestCase("a", "b")]
+        [TestCase(true, false)]
```

Comment:
Could we add some more cases for the non-trivial types like TimeSpan and DateTimeOffset?
Because it is these types that we are having issues on in the issues this PR is trying to fix.

Reply:
We have E2E test for TimeSpan and DateTimeOffset. This test is actually for target framework purpose. But it is OK to add more.